react/features/video-quality/middleware.js

```diff
@@ -0,0 +1,41 @@
+// @flow
+
+import { CONFERENCE_JOINED } from '../base/conference/actionTypes';
+import { VIDEO_QUALITY_LEVELS } from '../base/conference/constants';
+import { MiddlewareRegistry } from '../base/redux';
+
+import logger from './logger';
+
+/**
+ * Implements the middleware of the feature video-quality.
+ *
+ * @param {Store} store - The redux store.
+ * @returns {Function}
+ */
+MiddlewareRegistry.register(({ getState }) => next => action => {
+    const result = next(action);
+
+    switch (action.type) {
+    case CONFERENCE_JOINED: {
+        if (navigator.product === 'ReactNative') {
+            const state = getState();
+            const { conference, maxReceiverVideoQuality } = state['features/base/conference'];
+            const { resolution } = state['features/base/config'];
+
+            if (typeof resolution !== 'undefined') {
+                if (resolution < VIDEO_QUALITY_LEVELS.LOW) {
+                    logger.error(`Invalid frame height for video quality - ${resolution}`);
+
+                    break;
+                }
+                conference.setReceiverVideoConstraint(Math.min(resolution, maxReceiverVideoQuality));
```

Comment:
Don't we want to set the senderVideoConstraint as well ? I thought we wanted to send up to 360p on the mobile clients.

Reply:
That is done via config. Setting the sender constraint doesn’t work on mobile.

Reply:
oh right, it works only for p2p case and not the JVB case on RN.